src/app/modules/main/communities/tokens/module.nim

```diff
@@ -107,6 +107,14 @@ proc getTokenAndAmountList(self: Module, communityId: string, tokensJsonString:
   except Exception as e:
     error "Error getTokenAndAmountList", msg = e.msg
 
+proc getOwnerAndMasterTokens(self: Module, communityId: string): (CommunityTokenDto, CommunityTokenDto, bool) =
+  let ownerToken = self.controller.getOwnerToken(communityId)
+  let masterToken = self.controller.getTokenMasterToken(communityId)
```

Comment:
These functions call status-go and the DB, so they can make the app freeze. 

Either we should use only the community metadata (I think the addresses are there?), or we should make the whole flow async

Reply:
@jrainville I fixed this and I am getting addresses from permissions now.